File: my_py_pkg/my_py_pkg/hardware_status_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from my_robot_interfaces.msg import HardwareStatus 


class HardwareStatusPublisherNode(Node): # MODIFY NAME
    def __init__(self):
        super().__init__("hardware_status_publisher") # MODIFY NAME 
        self.publisher_ = self.create_publisher(HardwareStatus, "hardware_status", 10) # MODIFY TOPIC NAME
        self.timer = self.create_timer(1.0, self.publish_hardware_status)
        self.get_logger().info("Hardware Status Publisher Node has been started.")
        
    def publish_hardware_status(self):
        msg = HardwareStatus()
        msg.are_motors_ready = True
        msg.temperature = 75.0
        msg.debug_message = "All systems operational"
        self.publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = HardwareStatusPublisherNode() # MODIFY NAME
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/hardware_status_publisher.cpp
#include "my_robot_interfaces/msg/hardware_status.hpp"
#include "rclcpp/rclcpp.hpp"

using namespace std::chrono_literals;

class HardwareStatusPublisherNode : public rclcpp::Node {
public:
  HardwareStatusPublisherNode() : Node("hardware_status_publisher") {
    pub_ = this->create_publisher<my_robot_interfaces::msg::HardwareStatus>(
        "hardware_status", 10);
    timer_ = this->create_wall_timer(
        1s, std::bind(&HardwareStatusPublisherNode::timer_callback, this));
  }

private:
  rclcpp::Publisher<my_robot_interfaces::msg::HardwareStatus>::SharedPtr pub_;
  rclcpp::TimerBase::SharedPtr timer_;
  void timer_callback() {
    auto message = my_robot_interfaces::msg::HardwareStatus();
    message.temperature = 25.0;      // Example temperature value
    message.are_motors_ready = true; // Example motor status
    message.debug_message = "All systems operational"; // Example debug message
    pub_->publish(message);
  }
};

int main(int argc, char **argv) {
  rclcpp::init(argc, argv);
  auto node = std::make_shared<HardwareStatusPublisherNode>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}
